### ISSUE TITLE
Upgrade to TFS Aggregator 2.2 (Release Candidate 1) 

### ISSUE THREAD

**Issue by einar20** [tfsaggregator/tfsaggregator#199]
My company is upgrading TFS Server to version 2017 tonight.
Seems like I will need to upgrade TFS Aggregator to version 2.2 to support the new TFS Server.

Is ReleaseCandidate 1 for version 2.2 a "stable" version to run?

As far as I can see from the wiki, I will not have to update any of my existing scripts, so I assume this will be a rather smooth migration?

**Comment by giuliov** [comment]
We labeled RC due to limited testing, it may be that you find issues, while I hope you won't find any.
As per compatibility, we strive to avoid breaking existing rules.
Be careful of issue #112 in case you used the MSI: uninstall and reinstall the RC version.

Let us know how your upgrade goes.

**Comment by einar20** [comment]
OK, thanks. I'll let you know if I find any issues the next few days, or if everything continues to run smooth :-)

**Comment by einar20** [comment]
TFS is now upgraded to v 2017, and Aggregator 2.2 RC1 installed.
All scripts previously running on TFS 2015/Aggregator 2.0, still works fine and does their magic.

No issues to report :-) If I find something later, I will report back

**Comment by einar20** [comment]
Unfortunately we did end up with some issues during our first day with TFS 2017 & Aggregator 2.2 RC1.
Aggregator suddenly just stopped responding, and no aggregations were made on work items saved.

Invoking the same policy file from ConsoleApp, did however work just fine.

We managed to get Aggregator plugin work again, simply by re-copy file TFSAggregator2.ServerPlugin.policies to the TFS plugin folder (C:\Program Files\Microsoft Team Foundation Server 15.0\Application Tier\Web Services\bin\Plugins)

After a short while (10 - 30 minutes), Aggregator plugin however stopped responding again.
Again, re-copying policy file to TFS plugin folder would help - but only for a while.

We have re-installed TFS Aggregator2.2 and rebootet TFS server now. Reinstalled by:
- manually removing the plugin-DLLs
- un-install from MSI wizard
- install from MSI wizard

Problem still exists though - after a while, Aggregator plugin simply stops working.
Seems like TFS is unable to "hook up" properly to the Aggregator plugin

**Comment by giuliov** [comment]
If it works for a while then stops, should be some kind of locking or resource exhaustion. Do you have a log, EventLog message, some performance counters that helps spotting the source?

**Comment by einar20** [comment]
Theres no performance issues on the server.

However, EventLog has several entries for TFSAggregator.
Unfortunately my policy-file has logging level "Normal", so there's no details about the cause of error in eventlog, only headerinfo (ID=47513 - severity=Error - source=TFSAggregator)

I'll change log level to "verbose" or "diagnostic", and see if that will produce more information to the application eventlog

**Comment by einar20** [comment]
Seems like problem's solved :-) 

At least TFSAggregator has been running and doing it's magic for the last 4 hours now.
Previously, it would run for 1 hour at most, until it stopped responding.

What seems to have been the root-cause of my problems, is that there were files from TFS 2015 left on the server after upgrading to TFS 2017.
- The folder «C:\Program Files\Microsoft Team Foundation Server 14.0» was still present on server
- This folder also contained the "Plugin" subfolder, where TFSAggregator policy files from TFS 2015 still existed

After I removed the old policy files from "\Microsoft Team Foundation Server 14.0\Application Tier\Web Services\bin\Plugins", Aggregator has behaved as expected.


BTW: I also did some minor changes to my policy file, which I assume has no significance to this case though:
- changed "authentication autoImpersonate" from true to false
- removed code for a rule which ruleref was commented out in policy - as I noticed this generated a warning in output from ConsoleApp.

Changing logging level to "Verbose", didn't have any effect on eventlog output, I have asked system administrator to install "DebugView" on server, to be able to hook up on trace messages :)
